nav2_amcl/CMakeLists.txt

```diff
@@ -62,10 +62,12 @@ target_include_directories(motions_lib
   PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
-target_link_libraries(motions_lib PUBLIC
```

Comment:
For all CMakeLists:

We usually set the `dependencies` variable that contains all of the ament dependencies, which we use with `
ament_target_dependencies` and `ament_export_dependencies` to remove duplication. I think that is important to use here as well so that we don't miss important things (for example, this file misses `ament_export_dependencies`)

https://github.com/ros-navigation/navigation2/blob/humble/nav2_amcl/CMakeLists.txt#L53-L68